src/libraries/Microsoft.PowerFx.Core/Public/Types/VoidType.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Types
+{
+    /// <summary>
+    /// Special Type which can't be coerced or accepted by any types.
+    /// </summary>
+    public sealed class VoidType : FormulaType
+    {
+        internal VoidType()
+            : base(DType.Void)
+        {
+        }
+
+        /// <inheritdoc />
+        public override void Visit(ITypeVisitor visitor) => visitor.Visit(this);
```

Comment:
In what instances would we ever want the IR context from a void value? Shouldn't this class be a singleton with a single canonical instance (the Unit) with a NotInSource IRContext?

Reply:
Can we not include an IRContext in the constructor then, and pass a NotInSource into the base class constructor?

Reply:
Sorry this comment is in the wrong file. It should be down in the VoidValue file.